packages/hash/frontend/src/blocks/page/PageTitle/PageTitle.tsx

```diff
@@ -0,0 +1,122 @@
+import { Typography } from "@mui/material";
+import {
+  ChangeEventHandler,
+  FocusEventHandler,
+  KeyboardEventHandler,
+  useRef,
+  FunctionComponent,
+} from "react";
+import { useBlockProtocolUpdateEntity } from "../../../components/hooks/blockProtocolFunctions/useBlockProtocolUpdateEntity";
+import { rewriteEntityIdentifier } from "../../../lib/entities";
+import { usePageContext } from "../PageContext";
+import { cleanUpTitle, focusEditorBeginning, isValidPageTitle } from "./utils";
+
+type PageTitleProps = {
+  accountId: string;
+  entityId: string;
+  value: string;
+};
+
+// TODO: Add read-only mode based on page permissions
+export const PageTitle: FunctionComponent<PageTitleProps> = ({
+  accountId,
+  entityId,
+  value,
+}) => {
+  // TODO: Display update error once expected UX is discussed
+  const { updateEntity, updateEntityLoading } =
+    useBlockProtocolUpdateEntity(true);
+  const { editorView } = usePageContext();
+
+  const titleValueRef = useRef(value);
+
+  const handleInputChange: ChangeEventHandler<HTMLHeadingElement> = (event) => {
+    const { currentTarget } = event;
+    titleValueRef.current = currentTarget.textContent || "";
+
+    /**
+     * this prevents two issues
+     * 1 - when user pastes styled stuff into title
+     * other way of achieving this is implementing https://htmldom.dev/paste-as-plain-text/
+     * this looks like working well for now on all browser.
+     * we can switch to paste-as-plain-text later if we decide this solution is not enough
+     * 2 - firefox & safari renders a `br` tag when user clears the editable area,
+     * which prevents css selector `:empty` from working. This removes the `br` as well
+     */
+    if (currentTarget?.children.length) {
+      currentTarget.innerText = currentTarget.textContent || "";
```

Comment:
I think this causes the cursor to jump back to the start of the title when you paste in to it – maybe that's not a concern but thought worth flagging from a UX perspective. @jtewright @nonparibus

Reply:
If it does do that then we definitely want to fix it, as it's unexpected behavior.

Reply:
Should be fixed here https://github.com/hashintel/hash/pull/956/commits/eaafe1f4900ad2d20b3f01bafd5465f05ebaa7e1